### PR TITLE
Add Total mutations as clinical variable in Plots tab

### PR DESCRIPTION
**Pull request by oplantalech**
# What? Why?
Fix cbioportal/cbioportal#4291 .

Changes proposed in this pull request:
- a
- b

# Checks
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). If you are part of the
cBioPortal organization, notify the approprate team (remove inappropriate):

@cBioPortal/frontend

If you are not part of the cBioPortal organization look at who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them here:

### CODE REVIEW

File: src/pages/resultsView/plots/PlotsTab.tsx [hunk 1]
```diff
@@ -403,21 +403,44 @@ export default class PlotsTab extends React.Component<IPlotsTabProps,{}> {
 
     @computed get clinicalAttributeIdToClinicalAttribute():{[clinicalAttributeId:string]:ClinicalAttribute} {
         if (this.props.store.clinicalAttributes.isComplete) {
-            return this.props.store.clinicalAttributes.result.reduce((map:{[clinicalAttributeId:string]:ClinicalAttribute}, next)=>{
+            let _map: {[clinicalAttributeId: string]: ClinicalAttribute} = this.props.store.clinicalAttributes.result.reduce((map:{[clinicalAttributeId:string]:ClinicalAttribute}, next)=>{
```
Comment (adamabeshouse) [inline src/pages/resultsView/plots/PlotsTab.tsx]: you can shorten this as `_.keyBy(this.props.store.clinicalAttributes.result, c=>c.clinicalAttributeId)`

File: src/pages/resultsView/plots/PlotsTab.tsx [hunk 2]
```diff
@@ -403,21 +403,44 @@ export default class PlotsTab extends React.Component<IPlotsTabProps,{}> {
...
                 map[next.clinicalAttributeId] = next;
                 return map;
             }, {});
+            if (this.props.store.studyIds.isComplete) {
+                _map["total_mutations"] = {
```
Comment (adamabeshouse) [inline src/pages/resultsView/plots/PlotsTab.tsx]: its better to make this an enum value or something (look at the enum `SpecialAttribute`), at least a constant in this file `const TOTAL_MUTATIONS_CLIN_ATTR_ID = "total_mutations"`, then reference that elsewhere below

**Comment by jjgao** [comment]
@adamabeshouse @oplantalech the interface improvement discussed here (https://github.com/cBioPortal/cbioportal/issues/4432) may impact this PR.

**Comment by jjgao** [comment]
@oplantalech ok to point to release-1.5.0 assuming the current rc (and this pull request) does not contain other new code.

I was also asking in issue #4291 whether you would be able to add Fraction of Genome Altered as well?

**Comment by jjgao** [comment]
I would also like to test it before merging.

**Comment by oplantalech** [comment]
@jjgao Ok, I'll move it into release-1.15.0. Regarding the fraction of genome altered, it's almost ready, once this PR is merged I'm going to open a new one implementing this feature.

**Comment by jjgao** [comment]
Tested the feature. It looks great 👍

**Comment by jjgao** [comment]
Why are the e2e tests failing?

**Comment by inodb** [comment]
@jj	Seems like Total Mutations is now shown in plots tab instead of Diagnosis Age attribute: https://10779-66571349-gh.circle-artifacts.com/0/imageCompare.html. Also, seems like it might be better to log scale total mutations there. Prolly outside of the scope of this pr tho

**Comment by adamabeshouse** [comment]
@inodb it wouldnt be very hard but yeah probably out of scope of this PR

**Comment by jjgao** [comment]
I think we should update the screenshot and merge this pr. Log scale can be addressed in another pr.

**Comment by inodb** [comment]
@adamabeshouse do u know how the default clinical attribute on plots can be adjusted? Maybe better to keep using Diagnosis Age such that the plots look good. I guess it has to do with the order of clinical attributes in the dropdown menu of oncoprint?

**Comment by adamabeshouse** [comment]
@inodb thats right it is about the order

**Comment by inodb** [comment]
ideally test there is adjusted to select `Diagnosis Age` in the plots tab instead of just the first clinical attribute in the list. The total mutation count plots don't really show the whiskers very well, so kinda hard to spot regressions in that part of the code

**Comment by adamabeshouse** [comment]
@oplantalech you can use this command in `screenshot.spec.js`, in `it("plots tab clinical vs molecular"` between `broser.click(..)` and `waitForAndCheckPlotsTab`:

`browser.execute(function() { resultsViewPlotsTab.onHorizontalAxisClinicalAttributeSelect({ value: "AGE" }); });`

**Comment by adamabeshouse** [comment]
@oplantalech ah, okay so do the exact same thing in `it("plots tab clinical vs clinical boxplot"` and that should fix it

**Pull request merged**